exchanges/btcmarkets/btcmarkets.go

```diff
@@ -52,316 +75,472 @@ type BTCMarkets struct {
+		common.EncodeURLValues(btcMarketsOrders, params), nil, &resp)
+}
 
-		for x := range resp.Orders[i].Trades {
-			resp.Orders[i].Trades[x].Fee /= common.SatoshisPerBTC
-			resp.Orders[i].Trades[x].Price /= common.SatoshisPerBTC
-			resp.Orders[i].Trades[x].Volume /= common.SatoshisPerBTC
+// CancelOpenOrders cancels all open orders unless pairs are specified
+func (b *BTCMarkets) CancelOpenOrders(marketIDs []string) ([]CancelOrderResp, error) {
+	var resp []CancelOrderResp
+	req := make(map[string]interface{})
+	if len(marketIDs) > 0 {
+		var strTemp string
+		for x := range marketIDs {
+			strTemp += fmt.Sprintf("marketId=%s&", marketIDs[x])
 		}
+		req["marketId"] = strTemp[:len(strTemp)-1]
 	}
-	return resp.Orders, nil
+	return resp, b.SendAuthenticatedRequest(http.MethodDelete, btcMarketsOrders, req, &resp)
 }
 
-// GetAccountBalance returns the full account balance
-func (b *BTCMarkets) GetAccountBalance() ([]AccountBalance, error) {
-	var balance []AccountBalance
+// FetchOrder finds order based on the provided id
+func (b *BTCMarkets) FetchOrder(id string) (OrderData, error) {
+	var resp OrderData
+	return resp, b.SendAuthenticatedRequest(http.MethodGet, btcMarketsOrders+"/"+id,
+		nil, &resp)
+}
 
-	err := b.SendAuthenticatedRequest(http.MethodGet, btcMarketsAccountBalance, nil, &balance)
-	if err != nil {
-		return nil, err
-	}
+// RemoveOrder removes a given order
+func (b *BTCMarkets) RemoveOrder(id string) (CancelOrderResp, error) {
+	var resp CancelOrderResp
+	return resp, b.SendAuthenticatedRequest(http.MethodDelete, btcMarketsOrders+"/"+id,
+		nil, &resp)
+}
 
-	// All values are returned in Satoshis, even for fiat currencies.
-	for i := range balance {
-		balance[i].Balance /= common.SatoshisPerBTC
-		balance[i].PendingFunds /= common.SatoshisPerBTC
-	}
-	return balance, nil
+// ListWithdrawals lists the withdrawal history
+func (b *BTCMarkets) ListWithdrawals() ([]TransferData, error) {
+	var resp []TransferData
+	return resp, b.SendAuthenticatedRequest(http.MethodGet, btcMarketsWithdrawals, nil, &resp)
 }
 
-// GetTradingFee returns the account's trading fee for a currency pair
-func (b *BTCMarkets) GetTradingFee(base, quote currency.Code) (TradingFee, error) {
-	var tradingFee TradingFee
-	path := fmt.Sprintf(btcMarketsTradingFee, base, quote)
-	return tradingFee, b.SendAuthenticatedRequest(http.MethodGet, path, nil, &tradingFee)
+// GetWithdrawal gets withdrawawl info for a given id
+func (b *BTCMarkets) GetWithdrawal(id string) (TransferData, error) {
+	var resp TransferData
+	return resp, b.SendAuthenticatedRequest(http.MethodGet, btcMarketsWithdrawals+"/"+id,
+		nil, &resp)
 }
 
-// WithdrawCrypto withdraws cryptocurrency into a designated address
-func (b *BTCMarkets) WithdrawCrypto(amount float64, currency, address string) (string, error) {
-	newAmount := int64(amount * float64(common.SatoshisPerBTC))
+// ListDeposits lists the deposit history
+func (b *BTCMarkets) ListDeposits() ([]TransferData, error) {
```

Comment:
This seems to be missing optional parameter support:
https://api.btcmarkets.net/doc/v3#tag/Fund-Management-APIs/paths/~1v3~1deposits/get